addons/test_mail/tests/test_message_post.py

```diff
@@ -349,3 +349,28 @@ def test_post_post_w_template(self):
             subject='About %s' % test_record.name,
             body_content=test_record.name,
             attachments=[('first.txt', b'My first attachment', 'text/plain'), ('second.txt', b'My second attachment', 'text/plain')])
+
+    def test_notify_prepare_template_context_company_value(self):
+        companyA = self.env.company
+        companyB = self.env['res.company'].create({'name': 'Company B'})
+
+        record = self.env['res.partner'].sudo().with_context(self._test_context).create({
+            'name': 'Test',
+            'company_id': False,
+        })
+        record = record.with_context(mail_notrack=False)
```

Comment:
Could be great to use @users decorator and ensure its company is the right one (probably setting another one than default one to check it is correctly taken into account)

Reply:
Updated :+1: hope it is what you think...